x11-misc/slim/slim-1.3.9-r1.ebuild

```diff
@@ -0,0 +1,122 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake pam systemd
+
+DESCRIPTION="Simple Login Manager resurrected"
+HOMEPAGE="https://slim-fork.sourceforge.io/"
+SRC_URI="mirror://sourceforge/project/${PN}-fork/${P}.tar.gz"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~sparc ~x86"
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="branding pam"
+
+RDEPEND="x11-libs/libXmu
+	x11-libs/libX11
+	x11-libs/libXpm
+	x11-libs/libXft
+	x11-libs/libXrandr
+	media-libs/libpng:0=
+	media-libs/libjpeg-turbo
+	x11-apps/sessreg
+	pam? ( sys-libs/pam )"
+DEPEND="${RDEPEND}
+	x11-base/xorg-proto"
+BDEPEND="virtual/pkgconfig"
+PDEPEND="branding? ( >=x11-themes/slim-themes-1.2.3a-r3 )"
+
+PATCHES=(
+	# Our Gentoo-specific config changes
+	"${FILESDIR}"/${P}-config.diff
+	"${FILESDIR}"/${P}-greeter-session.diff
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	if use branding; then
+		sed -i -e '/current_theme/s/default/slim-gentoo-simple/' slim.conf || die
+	fi
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DUSE_PAM=$(usex pam)
+		-DUSE_CONSOLEKIT=OFF
+	)
+
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+
+	if use pam ; then
+		pamd_mimic system-local-login slim auth account session
+		pamd_mimic system-local-login slimlock auth
+	fi
+
+	systemd_dounit slim.service
+
+	insinto /usr/share/slim
+	newins "${FILESDIR}/Xsession-r3" Xsession
+
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}/slim.logrotate" slim
+
+	dodoc xinitrc.sample ChangeLog README TODO THEMES
+}
+
+pkg_postinst() {
+	# massage ${REPLACING_VERSIONS} to come up with whether or not it's a
+	# new install or if it's older than 1.3.2-r7
+	# Note - there should only ever be zero or one version as this package
+	# isn't slotted, so the logic doesn't worry about what happens if there
+	# are two, due to the case where a previous emerge attempt failed in
+	# the middle of qmerge.
+	local rv=none
+	for rv in ${REPLACING_VERSIONS} ; do
+		if ver_test "1.3.2-r7" -le "${rv}" ; then
+			rv=newer
+			break;
+		fi
+
+		if ver_test "1.0" -le "${rv}"  ; then
+			rv=older
+			break;
+		fi
+	done
+
+	if [[ ${rv} == none ]]; then
```

Comment:
You can use the `readme.gentoo-r1` eclass to display a message only on first installation, to skip this `none` check logic altogether.

Reply:
(this is optional - we can leave it for another time)